[채팅방] Infinite scroll 기능 추가

Junow opened the pull request:
# 채팅방에 infinite scroll 기능이 추가되었습니다. 

일단 머지를 해야할 시기가 다가와서 pr 을 올렸지만 지금 스크롤이 생기지 않는 매우 적은 양의 채팅만 있는 채팅방의 경우 무한 `dispatch` 가 이루어지고 있습니다. 이점 알아주시고 문제해결을 위해 데이터베이스 수정이 필요해서 같이 의논후 고치도록 하겠습니다.



- 1c7d0b7
	- socker url 받아오는 방식 변경
- f801455
	- getChat API 에 metaInfo.sender에 imageUrl 추가
- be11381
	- 채팅방에서 상대방 클릭시 profile popup 이 열림
![image](https://user-images.githubusercontent.com/32632221/84909373-8fd59400-b0f0-11ea-92c2-eca1f2ae7325.png)
- a4c1adc
	- 채팅과 채팅 사이 날짜 구분선 추가
![image](https://user-images.githubusercontent.com/32632221/84909443-aa0f7200-b0f0-11ea-892a-c6733e9cfa1c.png)

- 4c0a540
	- 뒤로가기 버튼 수정
![image](https://user-images.githubusercontent.com/32632221/84919173-8c480a00-b0fc-11ea-9b36-61911178a062.png)

- b027126
	- 채팅방 헤더 부분 CSS 수정 (수직 가운데 정렬, 방제목 수평 가운데 정렬)
![image](https://user-images.githubusercontent.com/32632221/84919338-c4e7e380-b0fc-11ea-913a-529e7e74c0a1.png)

- aad1aab
	- 메뉴버튼 추가
![image](https://user-images.githubusercontent.com/32632221/84919226-9e29ad00-b0fc-11ea-99f8-201b0e268f2e.png)

- 55076ec
	- 채팅방 헤더 타이틀 부분 가운데 정렬 (left, transform 처리)

- 86bf607
	- 에러메시지 추가
	- (잘못된 요청입니다)
	- code: 400

- a8e1b2a
	- 채팅메시지 일정 개수만큼 잘라서 가져올 수 있도록 chatApi 변경
- d6ea62a
	- 검색창 추가 (기능 아직 없음)
![image](https://user-images.githubusercontent.com/32632221/85427558-df1b3900-b5b6-11ea-9d17-e46f710b2963.png)

- 97c0be9
	- 채팅방에 infinite scroll 이 추가
	- 개선할점 issue 에 기록

- 580b343
	- chatroom component 의 Header 부분 불필요한 렌더링 `React.memo` 로 최적화 적용
- 6485299
	- chatroom component 의 TextArea 부분 따로 분리하였음
- af9bf2f
	- chat room container 가 삭제되었음

Junow committed:
Fix socket url by NODE_ENV

Junow committed:
Modify getChats - imageUrl 추가

Junow committed:
Add profile pop up in chat room

Junow committed:
Add date divider in chat room

Junow committed:
Add icon at back button in chat room

Junow committed:
Modify chat room header css

Junow committed:
Add menu icon in chat room header

Junow committed:
Modify css in chat room header

Junow committed:
Fix Router to Switch

Junow committed:
Fix child list key prop error

bbumjun reviewed (approved):
많은 기능들이 개선되었군요. 수고하셨습니다

Junow committed:
Add chat room search bar skeleton

Junow committed:
Add server error message

/**
 * @code
 * NOT FOUND (400)
 * @message
 * 잘못된 요청입니다.
 */

Junow committed:
Modify get chat api

querystring 에 offset, limit 추가

Junow committed:
Modify 채팅방 searchbar

검색바 나올 때 애니메이션 제거
디자인 수정

Junow committed:
Modify redux, redux-sage for chat

모든 채팅메시지 가져오지 않고 일정 개수 만큼 잘라서 요청하도록 변경

Junow committed:
Delete log

Junow committed:
Add IntersectionObserver hooks

Junow committed:
Fix onIntersec condition in chat infinite scroll

맨 처음 컴포넌트 mount 됬을때도 onIntersect 가 실행되어서
entry.time 이 5000 이상이어야 한다는 조건추가(entry 가 노출된 시간 )

Junow committed:
Modify redux chat status

add offset, limit field in redux chat status

Junow committed:
Add offset limit in chat saga

Junow committed:
Add infinite scroll in chat room

Junow committed:
Refactor chat room container

Header 컴포넌트 분리

Junow committed:
Refactor Header Component Memoized by React.memo

Junow committed:
Refactor Chatroom Component

TextArea 분리

Junow committed:
Chore 코드정리

Junow committed:
Refactor Chat Room Component

ChatArea 분리

Junow committed:
Delete ChatRoomContainer

chat room container 삭제

Junow committed:
Chore 코드정리

Junow committed:
Refactor Chat Room Component

ChatList 분리

Junow committed:
Fix useIntersectionObserver

observer target 조건부렌더링 하지 않는 것으로 ref 초기화시 observing 해제되는 문제 해결

Junow committed:
Merge branch 'develop' of https://github.com/Lee-Park-Bae-Project/KAKIOTALK into feature/room-func
